internal/core/src/index/ScalarIndex.h

```diff
@@ -50,6 +50,27 @@ class ScalarIndex : public IndexBase {
     virtual const TargetBitmap
     In(size_t n, const T* values) = 0;
 
+    virtual const TargetBitmap
+    InV2(size_t n,
+         const T* values,
+         const std::function<bool(size_t /* offset */)>& filter) {
+        auto res = In(n, values);
+        for (size_t offset = 0; offset < res.size(); offset++) {
+            res[offset] = res[offset] && filter(offset);
+        }
+        return res;
+    }
+
+    virtual void
+    InV3(size_t n,
+         const T* values,
+         const std::function<void(size_t /* offset */)>& callback) {
+        auto res = In(n, values);
+        for (size_t offset = 0; offset < res.size(); offset++) {
+            callback(offset);
+        }
+    }
+
```

Comment:
anyway, is that necessary add this interface in base class, other subclass need this function ?

Reply:
Yes, you are right that we cannot push down the filter or callback to all indexes. But I don't know how to design the interface, should we check whether the index is InvertedIndex?